src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1703,7 +1703,45 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                     }
                 }
                 value = ASR::down_cast<ASR::expr_t>(ASR::make_ConstantLogical_t(
-                    al, x.base.base.loc, result, source_type));
+                    al, x.base.base.loc, result, type));
+            } else if (ASRUtils::is_character(*source_type)) {
+                char* left_value = ASR::down_cast<ASR::ConstantString_t>(
+                                        ASRUtils::expr_value(left))->m_s;
```

Comment:
What is `ASRUtils::expr_value(left)` is nullptr? Is that case possible?

Reply:
Yes!